### PR TITLE
Unique questions stored in server

### PR DESCRIPTION
**Pull request by Kriti-bit**
## Related Issuse

Allowing only unique questions to be sent to the database.

Closes: #137 

#### Describe the changes you've made
I have made changes to the question model such that only unique questions are sent to the database.

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.

### CODE REVIEW

File: models/Questions.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 const mongoose = require('mongoose');
 
 const QuestionSchema = new mongoose.Schema({
-  question: { type: String, required: true },
+  question: { type: String, required: true, unique : true },
```
[the remark this answers is not in the document]
Comment (Kriti-bit) [reply models/Questions.js:4]: Got it. I'll make the necessary changes.
Thanks

**Comment by Kriti-bit** [comment]
Hi @aavishkarmishra, please review the PR and let me know if any changes are required.
Thanks

**Comment by Kriti-bit** [comment]
Hi @aavishkarmishra , please review the PR and let me know if any changes are required.
Thanks

File: routes/api/tests.js [hunk 2]
```diff
@@ -33,6 +33,16 @@ function GenerateTest(Questions, Rules) {
   return res;
 }
 
+function DuplicateQuestion(ques) {
+  var read_vals = Object.create(null);
+  for (var i = 0; i < ques.length; i++) {
+      var value = ques[i];
+      if (value in read_vals) {
```
Comment (aavishkarmishra) [inline routes/api/tests.js:40]: read_vals is always null.

Comment (Kriti-bit) [reply routes/api/tests.js]: I have changed the function, please review.
Thanks

**Comment by Kriti-bit** [comment]
Hi @aavishkarmishra, please let me know in case any other changes are required.
Thanks

**Review by aavishkarmishra — APPROVED** [review]

**Pull request merged**